devices/uart-16550/uarthw-pc.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * UART 16550 (generic PC)
+ *
+ * Copyright 2020, 2021 Phoenix Systems
+ * Author: Julia Kosowska, Pawel Pisarczyk, Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+#include <lib/errno.h>
+
+#include "uart-16550.h"
+
+
+typedef struct {
+	void *base;
+	unsigned int irq;
+} uarthw_context_t;
+
+
+static const uarthw_context_t uarts[] = {
+	{ (void *)(UART_COM1 | 0x1), UART_COM1_IRQ },
+	{ (void *)(UART_COM2 | 0x1), UART_COM2_IRQ },
+	{ (void *)UART_GALILEO1, UART_GALILEO1_IRQ },
+};
+
+
+unsigned char uarthw_read(void *hwctx, unsigned int reg)
+{
+	unsigned int addr = (unsigned int)((uarthw_context_t *)hwctx)->base;
```

Comment:
Declare this as volatile, remove cast on return

Reply:
Fixed.